libs/sysaudio/src/soundio.zig

```diff
@@ -264,18 +265,52 @@ pub fn requestDevice(self: Audio, allocator: std.mem.Allocator, config: DeviceDe
         .input => |d| d.handle.userdata = device,
         .output => |d| d.handle.userdata = device,
     }
-    var descriptor = config;
-    descriptor.mode = descriptor.mode orelse .output;
-    descriptor.channels = @intCast(u8, switch (handle) {
-        .input => |d| d.layout().channelCount(),
-        .output => |d| d.layout().channelCount(),
-    });
-    descriptor.sample_rate = @intCast(u32, switch (handle) {
-        .input => |d| d.sampleRate(),
-        .output => |d| d.sampleRate(),
-    });
-    std.log.info("channels {}", .{descriptor.channels.?});
-    std.log.info("sample_rate {}\n", .{descriptor.sample_rate.?});
+
+    // TODO(sysaudio): handle big endian architectures
+    const format: Format = switch (handle) {
+        .input => |d| switch (@intToEnum(SoundIoFormat, d.handle.format)) {
+            .U8 => .U8,
+            .S16LE => .S16,
+            .S24LE => .S24,
+            .S32LE => .S32,
+            .float32LE => .F32,
+            else => return error.InvalidParameter,
+        },
+        .output => |d| switch (@intToEnum(SoundIoFormat, d.handle.format)) {
+            .U8 => .U8,
+            .S16LE => .S16,
+            .S24LE => .S24,
+            .S32LE => .S32,
+            .float32LE => .F32,
+            else => return error.InvalidParameter,
+        },
+    };
+
+    // TODO(sysaudio): Get the device name
+    // const name_ptr = switch(handle) {
+    //     .input => |d| d.handle.name,
+    //     .output => |d| d.handle.name,
+    // };
+    // const name = std.mem.sliceTo(name_ptr, 0);
+    // std.log.info("name {s}", .{name});
+
+    var descriptor = DeviceDescriptor{
+        .is_raw = config.is_raw orelse false,
+        .format = format,
+        .mode = config.mode,
+        .id = std.mem.span(sio_device.handle.id),
+        .name = "",
+        .channels = @intCast(u8, switch (handle) {
+            .input => |d| d.layout().channelCount(),
+            .output => |d| d.layout().channelCount(),
+        }),
+        .sample_rate = @intCast(u32, switch (handle) {
+            .input => |d| d.sampleRate(),
+            .output => |d| d.sampleRate(),
+        }),
+    };
+    std.log.info("channels {}", .{descriptor.channels});
+    std.log.info("sample_rate {}\n", .{descriptor.sample_rate});
```

Comment:
Can you remove these log lines? I don't know why these are here, they shouldn't be.

Reply:
Probably some debugging prints that got forgotten